reference/test/solver/lower_trs.cpp

```diff
@@ -0,0 +1,155 @@
+
+
+#include <memory>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/exception.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+
+
+#include "core/test/utils/assertions.hpp"
+
+
+namespace {
+
+
+class LowerTrs : public ::testing::Test {
+protected:
+    using CsrMtx = gko::matrix::Csr<double, int>;
+    using Mtx = gko::matrix::Dense<>;
+    using Solver = gko::solver::LowerTrs<>;
+
+    LowerTrs()
+        : exec(gko::ReferenceExecutor::create()),
+          mtx(gko::initialize<Mtx>(
+              {{2, 0.0, 0.0}, {3.0, 1, 0.0}, {1.0, 2.0, 3}}, exec)),
+          b(gko::initialize<Mtx>({{2, 0.0, 0.0}}, exec)),
+          csr_mtx(gko::copy_and_convert_to<CsrMtx>(exec, mtx.get())),
+          lower_trs_factory(Solver::build().on(exec)),
+          lower_trs_solver(lower_trs_factory->generate(mtx, b))
+    {}
+
+    std::shared_ptr<const gko::Executor> exec;
+    std::shared_ptr<Mtx> mtx;
+    std::shared_ptr<Mtx> b;
+    std::shared_ptr<CsrMtx> csr_mtx;
+    std::unique_ptr<Solver::Factory> lower_trs_factory;
+    std::unique_ptr<Solver> lower_trs_solver;
+};
+
+
+TEST_F(LowerTrs, LowerTrsFactoryCreatesCorrectSolver)
+{
+    auto sys_mtx = lower_trs_solver->get_system_matrix();
+    auto d_sys_mtx = Mtx::create(exec);
+    sys_mtx->convert_to(gko::lend(d_sys_mtx));
+
+    ASSERT_EQ(lower_trs_solver->get_size(), gko::dim<2>(3, 3));
+    ASSERT_NE(sys_mtx, nullptr);
+    ASSERT_NE(lower_trs_solver->get_rhs(), nullptr);
+    GKO_ASSERT_MTX_NEAR(d_sys_mtx, mtx, 0);
+    ASSERT_EQ(lower_trs_solver->get_rhs(), b);
+}
+
+
+TEST_F(LowerTrs, CanBeCopied)
+{
+    auto copy = Solver::build().on(exec)->generate(Mtx::create(exec),
+                                                   Mtx::create(exec));
+    copy->copy_from(gko::lend(lower_trs_solver));
+
+    ASSERT_EQ(copy->get_size(), gko::dim<2>(3, 3));
+    auto copy_mtx = copy.get()->get_system_matrix();
+    auto d_copy_mtx = Mtx::create(exec);
+    copy_mtx->convert_to(gko::lend(d_copy_mtx));
+    auto copy_b = copy.get()->get_rhs();
+
+    GKO_ASSERT_MTX_NEAR(d_copy_mtx, mtx, 0);
+    GKO_ASSERT_MTX_NEAR(copy_b, b, 0);
+}
+
+
+TEST_F(LowerTrs, CanBeMoved)
+{
+    auto copy =
+        lower_trs_factory->generate(Mtx::create(exec), Mtx::create(exec));
+    copy->copy_from(std::move(lower_trs_solver));
+
+    ASSERT_EQ(copy->get_size(), gko::dim<2>(3, 3));
+    auto copy_mtx = copy.get()->get_system_matrix();
+    auto d_copy_mtx = Mtx::create(exec);
+    copy_mtx->convert_to(gko::lend(d_copy_mtx));
+    auto copy_b = copy.get()->get_rhs();
+
+    GKO_ASSERT_MTX_NEAR(d_copy_mtx, mtx, 0);
+    GKO_ASSERT_MTX_NEAR(copy_b, b, 0);
+}
+
+
+TEST_F(LowerTrs, CanBeCloned)
+{
+    auto clone = lower_trs_solver->clone();
+
+    auto clone_mtx = clone.get()->get_system_matrix();
+    auto d_clone_mtx = Mtx::create(exec);
+    clone_mtx->convert_to(gko::lend(d_clone_mtx));
+    auto clone_b = clone.get()->get_rhs();
```

Comment:
nit: I would kind of unify the use of `.get()` and `gko::lend` at least in this file, so you don't use both (here, you use both versions in consecutive lines).